Refactor loading UX

andresmgot opened the pull request:
<!--
 Before you open the request please review the following guidelines and tips to help it be more easily integrated:

 - Describe the scope of your change - i.e. what the change does.
 - Describe any known limitations with your change.
 - Please run any tests or examples that can exercise your modified code.

 Thank you for contributing!
 -->

### Description of the change

This PR includes several changes/improvements to the loading experience:

 - Use Clarity Core Component for the progress circle.
 - Improve style, adding some loading text explaining what's being loaded.
 - Fix behavior for the Application View where the resources were being requested several times, impacting the performance and the experience.
 - Some other minor CSS fixes.
 
![Peek 2021-03-17 11-51](https://user-images.githubusercontent.com/4025665/111456698-98a6d500-8717-11eb-8f6e-eb68726888e1.gif)

### Applicable issues

<!-- Enter any applicable Issues here (You can reference an issue using #) -->
  - fixes #2539
  - fixes #2276

andresmgot commented on dashboard/src/components/AppView/AppSecrets/AppSecrets.scss:
```diff
@@ -1,4 +1,6 @@
-  margin-right: 1.2rem;
+  margin-top: 1rem;
+  padding-right: 15px;
+  max-height: 15rem;
+  overflow-y: auto;
```
this was necessary (and related to https://github.com/kubeapps/kubeapps/issues/2462) when the app has too many secrets (e.g. harbor):

![Screenshot from 2021-03-17 12-39-05](https://user-images.githubusercontent.com/4025665/111462019-32718080-871e-11eb-9064-14d49e07be2d.png)

antgamdia commented:
Thanks for the gif, much clearer! I want to take a closer look at the PR later, however, just out of curiosity: have you changed the initial loading style (perhaps the one introduced by suspense? )

(This one:)
![image](https://user-images.githubusercontent.com/11535726/111461635-bc6d1980-871d-11eb-89b4-03f9b9ae8b48.png)

In other corporate websites using clarity I use to see this alternative approach:

![image](https://user-images.githubusercontent.com/11535726/111461724-ddce0580-871d-11eb-99ea-75d71e865199.png)

Something similar to: https://codesandbox.io/s/suspense-progress-bar-mu9lv

Or even this other one:

![image](https://user-images.githubusercontent.com/11535726/111462161-6187f200-871e-11eb-9f18-152f7c76cef9.png)



What do you think? Does the top spinner still make sense for the very initial loading?

andresmgot commented:
> What do you think? Does the top spinner still make sense for the very initial loading?

yes, I think it still makes sense. Let me try to find it and add at least a title like the case for TAC.

andresmgot commented:
> > What do you think? Does the top spinner still make sense for the very initial loading?
> 
> yes, I think it still makes sense. Let me try to find it and add at least a title like the case for TAC.

The `LoadingWrapper` from the `Suspense` component didn't appear (so I removed it) and the first one loaded was in the `ConfigLoader` so I added a title there:

![Screenshot from 2021-03-17 13-21-10](https://user-images.githubusercontent.com/4025665/111466756-f6d9b500-8723-11eb-8b47-e2bb93d07c3d.png)

antgamdia commented on dashboard/src/components/ConfigLoader/ConfigLoader.tsx:
```diff
@@ -18,7 +18,9 @@ function ConfigLoader({ getConfig, error, ...otherProps }: IConfigLoaderProps) {
     return <Alert theme="danger">Unable to load Kubeapps configuration: {error.message}</Alert>;
   }
-  return <LoadingWrapper {...otherProps} />;
+  return (
+    <LoadingWrapper className="margin-t-xxl" loadingText={<h1>Kubeapps</h1>} {...otherProps} />
+  );
```
> The `LoadingWrapper` from the `Suspense` component didn't appear (so I removed it) and the first one loaded was in the `ConfigLoader` so I added a title the

Oh much better now! Thank you! Minor thing here: since we are allowing modifying Kubeapps by other names, perhaps it should be the following for the sake of consistency:

```suggestion
  return (
    <LoadingWrapper className="margin-t-xxl" loadingText={<h1>{intl.formatMessage({ id: "Kubeapps", defaultMessage: "Kubeapps" })}</h1>} {...otherProps} />
  );
```

andresmgot replied:
The ConfigLoader is out of of the IntlProvider:

```
      <ConfigLoaderContainer>
        <ConnectedRouter history={history}>
          <Suspense fallback={null}>
            <IntlProvider
              locale={i18nConfig.locale}
              key={i18nConfig.locale}
              messages={i18nConfig.messages}
              defaultLocale="en"
            >
```

Causing an issue:

```
Error: [React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.
```

Should we move the IntlProvider up in the tree or leave the ConfigLoader out?

antgamdia replied:
Mmm, I don't remember right now, but since we are not using anything from the `ConfigLoaderConainer` in the `IntlProvider` I guess we can move it up.

antgamdia reviewed (approved):
Just minor comments and some doubts here. I've pulled the PR and it looks much better now, thank you for the UI changes of course, but thanks for the websocket thing! (since I'm with the Pinniped was issue, now it is easier to debug what's happening :)   )

andresmgot commented on dashboard/src/components/Catalog/Catalog.tsx:
```diff
@@ -447,7 +447,7 @@ function Catalog(props: ICatalogProps) {
                         filters[filterNames.TYPE].find((type: string) => type === "Charts")) && (
                         <div className="endPageMessage">
-                          <LoadingWrapper medium={true} loaded={false} />
+                          <LoadingWrapper loaded={false} />
```
yes, in this case, is the same loading animation when fetching new apps:

![Screenshot from 2021-03-17 17-02-42](https://user-images.githubusercontent.com/4025665/111498915-b3427380-8742-11eb-8235-cea1048da59d.png)

And I find it a bit distracting if there is text there that I am not fast enough to read.

antgamdia replied:
Ok then, thanks for checking!

antgamdia commented on dashboard/src/components/AppList/AppList.tsx:
```diff
@@ -123,7 +123,11 @@ function AppList() {
-      <LoadingWrapper loaded={!isFetching && !isFetchingResources}>
+      <LoadingWrapper
+        loaded={!isFetching && !isFetchingResources}
+        loadingText="Getting the list of applications..."
```
Great, I agree, a full capitalized sentence looks terrible, haha. Thanks for the change :)
